tests/test_notebook.py

```diff
@@ -0,0 +1,145 @@
+PROJECT_URL = "dummynamespace/dummyproject/0123456789"
+SERVER_NAME = "dummynames-dummyproje-0123456"
+NON_DEVELOPER_ACCESS = DEVELOPER_ACCESS - 1
+
+
+def test_can_check_health(client):
+    response = client.get("/health")
+    assert response.status_code == 200
+
+
+def test_can_create_notebooks(client):
+    response = client.post(f"/service/{PROJECT_URL}", headers=AUTHORIZED_HEADERS)
+    assert response.status_code == 200 or response.status_code == 201
+
+
+def test_can_get_created_notebooks(client):
+    client.post(f"/service/{PROJECT_URL}", headers=AUTHORIZED_HEADERS)
+
+    response = client.get("/service/servers", headers=AUTHORIZED_HEADERS)
+    assert response.status_code == 200
+    assert SERVER_NAME in response.json.get("servers")
+
+
+def test_can_get_server_status_for_created_notebooks(client, kubernetes_client):
+    response = client.post(f"/service/{PROJECT_URL}", headers=AUTHORIZED_HEADERS)
+
+    response = client.get(f"/service/{PROJECT_URL}", headers=AUTHORIZED_HEADERS)
+    assert response.status_code == 200
+    assert response.json.get("name") == SERVER_NAME
+
+
+def test_getting_notebooks_returns_nothing_when_no_notebook_is_created(client):
+    response = client.get("/service/servers", headers=AUTHORIZED_HEADERS)
+    assert response.status_code == 200
+    assert response.json.get("servers") == {}
+
+
+def test_can_get_server_options(client):
+    client.post(f"/service/{PROJECT_URL}", headers=AUTHORIZED_HEADERS)
+
+    response = client.get(
+        f"/service/{PROJECT_URL}/server_options", headers=AUTHORIZED_HEADERS
+    )
+    assert response.status_code == 200
+    assert response.json == {"dummy-key": {"default": "dummy-value"}}
+
+
+def test_can_get_pods_logs(client, kubernetes_client):
+    client.post(f"/service/{PROJECT_URL}", headers=AUTHORIZED_HEADERS)
+
+    headers = AUTHORIZED_HEADERS.copy()
+    headers.update({"Accept": "text/plain"})
+    response = client.get(f"/service/{PROJECT_URL}/logs", headers=headers)
+    assert response.status_code == 200
+
+
+def test_can_delete_created_notebooks(client):
+    client.post(f"/service/{PROJECT_URL}", headers=AUTHORIZED_HEADERS)
+
+    response = client.delete(f"/service/{PROJECT_URL}", headers=AUTHORIZED_HEADERS)
+    assert response.status_code == 204
+
+
+def test_can_delete_created_notebooks_with_server_name(client):
+    client.post(f"/service/{PROJECT_URL}", headers=AUTHORIZED_HEADERS)
+
+    response = client.delete(
+        f"/service/servers/{SERVER_NAME}", headers=AUTHORIZED_HEADERS
+    )
+    assert response.status_code == 204
+
+
+def test_recreating_notebooks_return_current_server(client, kubernetes_client):
+    response = client.post(f"/service/{PROJECT_URL}", headers=AUTHORIZED_HEADERS)
+
+    response = client.post(f"/service/{PROJECT_URL}", headers=AUTHORIZED_HEADERS)
+    assert response.status_code == 200
+    assert SERVER_NAME in response.json.get("name")
+
+
+@pytest.mark.parametrize("gitlab", [NON_DEVELOPER_ACCESS], indirect=True)
```

Comment:
why not parametrize `test_can_create_notebooks`?

Reply:
Then we would be testing two things (which are more or less different) in one test. I prefer to have two tests here.